torchaudio_contrib/layers.py

```diff
@@ -1,142 +1,241 @@
-
-
 import torch
+import math
 import torch.nn as nn
-import torch.nn.functional as F
 
-from .functional import spectrogram, create_mel_filter
+from .functional import stft_defaults, _stft, complex_norm, \
+    create_mel_filter, phase_vocoder
 
 
-class Spectrogram(nn.Module):
+class _ModuleNoStateBuffers(nn.Module):
     """
-    Module that outputs the spectrogram
-    of an audio signal with shape (batch, channel, time_hop, frequency_bins).
-
-    Its implemented as a layer so that the computation can be faster (done dynamically
-    on GPU) and no need to store the transforms. More information:
-        - https://github.com/keunwoochoi/kapre
-        - https://arxiv.org/pdf/1706.05781.pdf
+    Extension of nn.Module that removes buffers
+    from state_dict.
 
     Args:
-     * hop: int > 0
-       -  Hop length between frames in sample,  should be <= n_fft.
-       -  Default: None (in which case n_fft // 4 is used)
-     * n_fft: int > 0 
-       -  Size of the fft.
-       -  Default: 2048
-     * pad: int >= 0
-       -  Amount of two sided zero padding to apply.
-       -  Default: 0
-     * window: torch.Tensor,
-       -  Windowing used in the stft.
-       -  Default: None (in which case torch.hann_window(n_fft) is used)
-     * sr: int > 0
-       -  Sampling rate of the audio signal. This may not be the same in all samples (?)
-       -  Default: 44100
-     * power: int = 1,2,None
-       -  Power to normalize the spectrogram to, make None to work with complex stft
-       -  Default: 1
-     * kwargs: 
-       -  Any named arguments to be passed to the stft
+        buffers (dict): mapping of names and tensors to be registered.
+    """
 
+    def __init__(self, buffers):
+        super(_ModuleNoStateBuffers, self).__init__()
+        for k, v in buffers.items():
+            self.register_buffer(k, v)
+
+    def state_dict(self, destination=None, prefix='', keep_vars=False):
+        ret = super(_ModuleNoStateBuffers, self).state_dict(
+            destination, prefix, keep_vars)
+        for k in self._buffers.keys():
+            del ret[prefix + k]
+        return ret
```

Comment:
I think you will also need to override `_load_state_dict`. I have:
```python
    def state_dict(self, destination=None, prefix='', keep_vars=False):
        result = super(_ModuleNoStateBuffers, self).state_dict(destination, prefix, keep_vars)
        # remove all buffers; we use them as cached constants
        for k in self._buffers:
            del result[prefix + k]
        return result

    def _load_from_state_dict(self, state_dict, prefix, *args, **kwargs):
        # temporarily hide the buffers; we do not want to restore them
        buffers = self._buffers
        self._buffers = {}
        result = super(_ModuleNoStateBuffers, self)._load_from_state_dict(state_dict, prefix, *args, **kwargs)
        self._buffers = buffers
        return result
```

Reply:
Yes, absolutely!